ui/component/settingWalletServer/internal/inputRow.jsx

```diff
@@ -1,47 +1,63 @@
 // @flow
-import * as ICONS from 'constants/icons';
 import React, { useState, useEffect } from 'react';
 import Button from 'component/button';
-import { FormField } from 'component/common/form';
+import { Form, FormField } from 'component/common/form';
 
 type Props = {
-  update: (string) => void,
+  update: ([string, string]) => void,
 };
 
-const VALID_IPADDRESS_REGEX = new RegExp('^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\\.)){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$');
-const VALID_HOSTNAME_REGEX = new RegExp('^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(\\.))+([A-Za-z]|[A-Za-z][A-Za-z]*[A-Za-z])$');
+const VALID_IPADDRESS_REGEX = new RegExp(
+  '^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\\.)){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$'
+);
+const VALID_HOSTNAME_REGEX = new RegExp(
+  '^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(\\.))+([A-Za-z]|[A-Za-z][A-Za-z]*[A-Za-z])$'
+);
 const VALID_PORT_REGEX = new RegExp('^([0-9]){1,5}$');
 
 function ServerInputRow(props: Props) {
   const { update } = props;
-
   const [hostString, setHostString] = useState('');
   const [portString, setPortString] = useState('');
   const [validServerString, setValidServerString] = useState(false);
 
   useEffect(() => {
-    setValidServerString((VALID_IPADDRESS_REGEX.test(hostString) || VALID_HOSTNAME_REGEX.test(hostString)) && VALID_PORT_REGEX.test(portString));
+    setValidServerString(
+      (VALID_IPADDRESS_REGEX.test(hostString) || VALID_HOSTNAME_REGEX.test(hostString)) &&
+        VALID_PORT_REGEX.test(portString)
+    );
   }, [hostString, portString, validServerString, setValidServerString]);
 
-  function onClick() {
+  function onSubmit() {
     update([hostString, portString]);
     setHostString('');
     setPortString('');
   }
 
   return (
-    <tr>
-      <td>
-        <FormField type="text" value={hostString} onChange={e => setHostString(e.target.value)} />
-      </td>
-      <td>
-        <FormField type="text" value={portString} onChange={e => setPortString(e.target.value)} />
-      </td>
-      <td />
-      <td>
-        <Button button={'link'} icon={ICONS.ADD} disabled={!validServerString} onClick={onClick} />
-      </td>
-    </tr>
+    <Form onSubmit={onSubmit}>
+      <div className="section__actions">
+        <FormField
+          type="text"
+          label={__('Host')}
+          placeholder={'code.freezepeach.fun'}
+          value={hostString}
+          onChange={e => setHostString(e.target.value)}
+        />
+        <span className="form-field__conjuction">:</span>
+        <FormField
+          type="number"
+          label={__('Port')}
+          placeholder={'50001'}
+          value={portString}
+          onChange={e => setPortString(String(e.target.value))}
+        />
+      </div>
+
+      <div className="section__actions">
+        <Button type="submit" button="primary" label={__('Add')} disabled={!validServerString} />
```

Comment:
It is likely that at least some users will find it confusing that invalid input blocks the submission. If you have entered invalid input, you are given no feedback that it is invalid other than inability to submit.

I would almost prefer being able to submit but then being told I entered something invalid to blocking submission. Alternatively, an error styling could be applied to the inputs on blur.